lib/ui/painting.dart

```diff
@@ -1606,14 +1618,150 @@ class Image extends NativeFieldWrapperClass2 {
   /// Returns an error message on failure, null on success.
   String? _toByteData(int format, _Callback<Uint8List?> callback) native 'Image_toByteData';
 
+  bool _disposed = false;
   /// Release the resources used by this object. The object is no longer usable
   /// after this method is called.
-  void dispose() native 'Image_dispose';
+  ///
+  /// All outstanding handles from [createHandle] should be disposed before
+  /// calling this. Disposing all outstanding handles will automatically
+  /// dispose this object.
+  void dispose() {
+    assert(() {
+      assert(!_disposed);
+      assert(
+        _handles.isEmpty,
+        'Attempted to dispose of an Image object that has ${_handles.length} open handles.',
```

Comment:
Should this print the (de-duped?) stack traces of the open handles?

Reply:
I think that will become pretty overwhelming - for instance, in testing it's not terribly uncommon to have 4 or 5 open handles (between ImageCache in a couple spots, the ImageStream, the widget actually using the image). The stacks might be hundreds of lines long a piece.

In the Framework we can use the Diagnosticable stuff and some stack parsing logic to make it a bit more manageable via the debug property. Maybe we could make this error message suggest using that if you're hitting this?

Reply:
> Maybe we could make this error message suggest using that if you're hitting this?

Sounds good.

Reply:
This is now just an assertion that should never fire, unless we introduce a bug into `dart:ui`. I've updated the text to recommend filing a bug.